bcipy/tasks/rsvp/main_frame.py

```diff
@@ -253,3 +261,77 @@ def prepare_stimuli(self):
             timing=self.stimuli_timing,
             seq_constants=self.seq_constants)
         return stimuli
+
+
+class CriteriaEvaluator():
+    """Evaluates whether an epoch should commit to a decision based on the
+    provided criteria.
+    
+    Parameters:
+    -----------
+        continue_criteria: list of functions; if any of these evaluate to true the
+            decision maker continues.
+        commit_criteria: list of functions; if any of these return true and
+            continue_criteria are all false, decision maker commits to a decision.
+    """
+
+    def __init__(self, continue_criteria: list, commit_criteria: list):
```

Comment:
If we made an abstract `DecisionCriteria` with an `apply` method (not attached to the names here!), we could then extend it and place our business logic in `apply`. We could then explicitly declare what we expect in our list: `List[DecisionCriteria]`. It's an improvement to have them as methods, but a further abstraction could prove useful as we continue adding criteria.

Reply:
I considered doing this but didn't want to over-engineer things. But you make a good argument about the documentation aspect of this approach. I'll give it a try and see how it looks.

Reply:
I refactored into objects and I think it turned out well. Thanks for the suggestion.